src/overlays/actors/ovl_Boss_Ganon/z_boss_ganon.c

```diff
@@ -108,127 +98,4928 @@ static ColliderCylinderInit D_808E4C2C = {
+                BossGanon_SetupHitByLightBall(dorf, globalCtx);
+                this->timers[0] = 150;
+                numEffects = 40;
+                this->unk_1C2 = 1;
+                this->actor.speedXZ = 0.0f;
+            }
+            break;
+    }
+
+    if (this->unk_1C2 >= 0xB) {
+        xzDist = (this->unk_1C2 == 0xC) ? -65.0f : 0.0f;
+
+        if ((fabsf(this->actor.world.pos.x) > (465.0f + xzDist)) ||
+            (fabsf(this->actor.world.pos.z) > (465.0f + xzDist)) || ((this->actor.world.pos.y < 0.0f)) ||
+            (this->actor.world.pos.y > 450.0f)) {
+            this->unk_1C2 = 1;
+            this->actor.speedXZ = 0.0f;
+            numEffects = 10;
+            BossGanon_CheckFallingPlatforms(this, globalCtx, &this->actor.world.pos);
+            Actor_SpawnAsChild(&globalCtx->actorCtx, &dorf->actor, globalCtx, 0xE8, this->actor.world.pos.x,
+                               this->actor.world.pos.y, this->actor.world.pos.z, 0, 0, 0, 0x190);
+        }
+    }
+
+    if (numEffects) {
+        Audio_PlaySoundAtPosition(globalCtx, &this->actor.world.pos, 80, NA_SE_EN_FANTOM_THUNDER);
+
+        for (i = 0; i < numEffects; i++) {
+            sp60.x = Rand_CenteredFloat(30.0f);
+            sp60.y = Rand_CenteredFloat(30.0f);
+            sp60.z = Rand_CenteredFloat(30.0);
+
+            BossGanonEff_SpawnLightRay(globalCtx, &this->actor.world.pos, &sp60, &sZeroVec,
+                                       Rand_ZeroFloat(200.0f) + 500.0f, 15.0f, 0x1E);
+        }
+    }
+}
+
+static Gfx* sBigMagicLightStreakDLists[] = {
+    gDorfLightStreak12DL, gDorfLightStreak11DL, gDorfLightStreak10DL, gDorfLightStreak9DL,
+    gDorfLightStreak8DL,  gDorfLightStreak7DL,  gDorfLightStreak6DL,  gDorfLightStreak5DL,
+    gDorfLightStreak4DL,  gDorfLightStreak3DL,  gDorfLightStreak2DL,  gDorfLightStreak1DL,
+};
+
+void func_808E324C(Actor* thisx, GlobalContext* globalCtx) {
+    BossGanon* this = (BossGanon*)thisx;
+    Mtx* mtxArr;
+    s16 i;
+    s32 bodyPart;
+
+    mtxArr = Graph_Alloc(globalCtx->state.gfxCtx, 12 * sizeof(Mtx));
+
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_boss_ganon.c", 10489);
+
+    func_80093D84(globalCtx->state.gfxCtx);
+    gDPSetPrimColor(POLY_XLU_DISP++, 0, 0x80, 255, 255, 255, (s8)this->fwork[FWORK_1]);
+    gDPSetEnvColor(POLY_XLU_DISP++, 150, 255, 0, 128);
+    gSPSegment(POLY_XLU_DISP++, 0x0D, mtxArr);
+
+    for (i = 0; i < 12; i++) {
+        bodyPart = (s16)(((this->unk_1A6 - i) + 0xF) % 15);
+        Matrix_Translate(this->bodyPartsPos[bodyPart].x, this->bodyPartsPos[bodyPart].y, this->bodyPartsPos[bodyPart].z,
+                         MTXMODE_NEW);
+        Matrix_RotateY(this->unk_3C4[bodyPart].y, MTXMODE_APPLY);
+        Matrix_RotateX(-this->unk_3C4[bodyPart].x, MTXMODE_APPLY);
+        Matrix_Scale(this->actor.scale.x, this->actor.scale.y, this->actor.scale.z, MTXMODE_APPLY);
+        Matrix_RotateY(1.5707964f, MTXMODE_APPLY);
```

Comment:
```suggestion
        Matrix_RotateY(M_PI / 2, MTXMODE_APPLY);
```
and other `1.5707964f`